packages/vite/src/node/plugins/worker.ts

```diff
@@ -121,6 +125,33 @@ async function bundleWorkerEntry(
   return emitSourcemapForWorkerEntry(config, chunk)
 }
 
+const importRE = /import\(["'](.+?)["']\)/g
+function areFilesEqual(path1: string, path2: string) {
+  return path.basename(path1) === path.basename(path2)
+}
+
+function replaceDynamicImportsPlugin(config: ResolvedConfig): Plugin {
+  return {
+    name: 'rollup-plugin-replace-dynamic-imports-in-worker',
+    generateBundle(options, bundle) {
+      for (const [_, output] of Object.entries(bundle)) {
+        if (output.type === 'chunk' && output.isEntry === true) {
+          output.code = output.code.replace(importRE, (match, extractedUrl) => {
+            const importUrl = output.dynamicImports.find((url) =>
+              areFilesEqual(extractedUrl, url),
+            )
+            if (importUrl) {
+              return `import(self.location.origin + "${config.rawBase + importUrl}")`
```

Comment:
This approach might have an issue with relative base. I'm checking `pnpm -C playground/worker build:relative-base` + `preview:relative-base` and I see this code in the inline worker:

```ts
import(self.location.origin + "./worker-chunks/worker_chunk-vite-D2sZgpE4.js")

// TypeError: Failed to resolve module specifier 'http://localhost:4173./worker-chunks/worker_chunk-index-JyeH1t5q.js'
```

I'm not sure if there's any mechanism possible to actually handle this though. (Maybe main script needs to tell workers where the main script is and compute relative path based on that?)

Reply:
Can this be solved in this way? Use config.build.outDir to extract the path we need, if outDir = "dist" or './dist' use origin + / + url, if outDir = 'dist/es/' use origin + /es/.

Reply:
You're right, we can't get the full base path, maybe set it up from the main script.

Reply:
I think this is a tricky problem to solve in a general manner. For example, I think it's technically possible to have an inline worker inside inline worker, but I would imagine supporting that would cause more complication.
We need a discussion with a team, but it might be acceptable, for example, to limit the use case only for non relative base case to simplify things.

Either way, thanks for initiating a PR :+1: